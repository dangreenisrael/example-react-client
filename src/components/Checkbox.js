import React, {Component} from 'react';

class Checkbox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isChecked: props.isChecked || false
    };
    this.handleInputChange = this.handleInputChange.bind(this);
  }
  handleInputChange(event) {
    const value = event.target.checked;
    this.setState({
      isChecked: value
    });
    if (this.props.onChange){
      this.props.onChange(value);
    }
  }
  render() {
    const {name} = this.props;
    return (
        <input
            name={name}
            type="checkbox"
            checked={this.state.isChecked}
            onChange={this.handleInputChange} />
    );
  }

}

export default Checkbox;
