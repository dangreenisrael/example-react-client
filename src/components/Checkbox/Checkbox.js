import React, {Component} from 'react';

class Checkbox extends Component {
  constructor(props) {
    super(props);
    this.handleInputChange = this.handleInputChange.bind(this);
  }
  handleInputChange(event) {
    const value = event.target.checked;
    if (this.props.onChange){
      this.props.onChange(value);
    }
  }
  render() {
    const {name,checked} = this.props;
    const isIndeterminate = (checked === "indeterminate");
    return (
        <input
            name={name}
            type="checkbox"
            checked={checked}
            ref={elem => elem && (elem.indeterminate = isIndeterminate)}
            onChange={this.handleInputChange} />

    );
    /*
     * About the weird ref thing, See: https://github.com/facebook/react/issues/1798
     */
  }
}

export default Checkbox;
