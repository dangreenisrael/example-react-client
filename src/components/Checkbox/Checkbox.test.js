import React from 'react';
import renderer from 'react-test-renderer';
import Checkbox from './Checkbox';

test('Checkbox Renders Correctly', () => {
  const tree = renderer.create(
      <Checkbox onChange={()=>{}} checked={true} name="checkMe"/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});