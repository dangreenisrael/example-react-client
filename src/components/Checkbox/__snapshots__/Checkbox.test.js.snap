exports[`test Checkbox Renders Correctly 1`] = `
<input
  checked={true}
  name="checkMe"
  onChange={[Function]}
  type="checkbox" />
`;
